gxutil/get.go

```diff
@@ -30,12 +30,13 @@ func (pm *PM) GetPackageTo(hash, out string) (*Package, error) {
 		err := FindPackageInDir(&pkg, out)
 		if err == nil {
 			return &pkg, nil
-		} else if !os.IsNotExist(err) {
-			return nil, err
```

Comment:
Sorry, I'm not following, this return takes place if there was a different error than "couldn't find the JSON file in the `out` directory", why is it necessary to remove this? I mean, why would I care to remove a directory it there was any other error (which I'm suspecting would be more severe) than file not found?

Reply:
> Since that's not a "not exist" error, it would trigger this condition,

Yes, you're right. It seems that `LoadPackageFile` is used as a dual function (which is not very nice),

https://github.com/whyrusleeping/gx/blob/7e9bc0de5fe60ec96367ac17dcab79535cf91495/gxutil/get.go#L90-L100

the first call is assuming the `dir` argument has the path to `hash/pkg-name`, that call is allowed to silently fail and the second scenario is assumed, `dir` has the path only to `hash` and `PackageNameInDir` looks for a single directory inside of there that should have the `pkg-name` directory (and `LoadPackageFile` is called again with the found `pkg-name` to look for the JSON file). If `PackageNameInDir` doesn't find any directories inside it returns this custom error you're linking (I would rather it returned the standard `ErrNotExist`).

So yes, there are multiple possible returned errors. I would prefer to explicitly name them and ask for them in addition to the `IsNotExist()` check, instead of having the `RemoveAll` as a catch-all case, e.g., in the (very rare) case there were two directories inside of `hash`, let's say the `pkg-name` directory and some other spurious one, `PackageNameInDir` would also fail,

https://github.com/whyrusleeping/gx/blob/7e9bc0de5fe60ec96367ac17dcab79535cf91495/gxutil/get.go#L112-L114

and this would cause this patch to remove the entire `hash` directory (which wouldn't be the actual objective of the `RemoveAll` call I guess).

Anyway, that's just an idea to start tidying up the code in the future, but for practical purposes this PR works, so LGTM.

Reply:
So.. I inadvertently delete my previous review comment (ouch), the digest is: yes, you're right, there are multiple returned errors, LGTM.

Reply:
@schomatis 
I have retrieved your comment
***
>Since that's not a "not exist" error, it would trigger this condition

Yes, you're right. It seems that `LoadPackageFile` is used as a dual function (which is not very nice),

https://github.com/whyrusleeping/gx/blob/7e9bc0de5fe60ec96367ac17dcab79535cf91495/gxutil/get.go#L90-L100

the first call is assuming the `dir` argument has the path to `hash/pkg-name`, that call is allowed to silently fail and the second scenario is assumed, `dir` has the path only to `hash` and `PackageNameInDir` looks for a single directory inside of there that should have the `pkg-name` directory (and `LoadPackageFile` is called again with the found `pkg-name` to look for the JSON file). If `PackageNameInDir` doesn't find any directories inside it returns this custom error you're linking (I would rather it returned the standard `ErrNotExist`).

So yes, there are multiple possible returned errors. I would prefer to explicitly name them and ask for them in addition to the `IsNotExist()` check, instead of having the `RemoveAll` as a catch-all case, e.g., in the (very rare) case there were two directories inside of `hash`, let's say the `pkg-name` directory and some other spurious one, `PackageNameInDir` would also fail,

https://github.com/whyrusleeping/gx/blob/7e9bc0de5fe60ec96367ac17dcab79535cf91495/gxutil/get.go#L112-L114

and this would cause this patch to remove the entire `hash` directory (which wouldn't be the actual objective of the `RemoveAll` call I guess).

Anyway, that's just an idea to start tidying up the code in the future, but for practical purposes this PR works, so LGTM.